core/src/display_object/movie_clip.rs

```diff
@@ -1733,6 +1738,36 @@ impl<'gc> TDisplayObject<'gc> for MovieClip<'gc> {
         self.0.read().movie().version()
     }
 
+    fn enter_frame(&self, context: &mut UpdateContext<'_, 'gc, '_>) {
+        for child in self.iter_render_list() {
+            child.enter_frame(context);
+        }
+
+        if context.avm_type() == AvmType::Avm2 {
+            let is_playing = self.playing();
+
+            if is_playing {
+                // Frame destruction happens in-line with frame number advance.
+                // If we expect to loop, we do not run `RemoveObject` tags.
```

Comment:
This is talking about looping back to the first frame, not 'looping' on the current frame, right?